pkg/scheduler/api/node_info.go

```diff
@@ -295,17 +295,34 @@ func (ni *NodeInfo) setNodeGPUInfo(node *v1.Node) {
 
 // SetNode sets kubernetes node object to nodeInfo object
 func (ni *NodeInfo) SetNode(node *v1.Node) {
-	ni.setOversubscription(node)
 	ni.setNodeState(node)
-	ni.setNodeGPUInfo(node)
-	ni.setRevocableZone(node)
-
 	if !ni.Ready() {
-		klog.Warningf("Failed to set node info, phase: %s, reason: %s",
-			ni.State.Phase, ni.State.Reason)
+		klog.Warningf("Failed to set node info for %s, phase: %s, reason: %s",
+			ni.Name, ni.State.Phase, ni.State.Reason)
+		return
+	}
+
+	// Dry run, make sure all fields other than `State` are in the original state.
+	copy := ni.Clone()
```

Comment:
why clone?

Reply:
To make make sure all fields are left untouched and in the original state.

Reply:
Can you point where do we mutate this node?

Reply:
- The original codes are all changing fields of nodeInfo.
- Here, the `clone` i used, is merely a dry-run to make sure if all these changes makes this node not ready then will shall not to actually do this operation.